src/content/1.7/testing/web-acceptance-tests/how-to-contribute-and-create-web-acceptance-tests.md

```diff
@@ -0,0 +1,213 @@
+
+*nametype\* in `camelCase`.
+
+For example, a button used to submit the main form in the order page should be named: `submitMainFormButton`.
+
+
+#### Types
+Each selector must belong to a certain type. The whole list (not exhaustive) is here:
+- button
+- link
+- block
+- image
+- icon
+- text
+- modal
+- *other HTML elements...*
+
+
+- table
+- table-header
+- table-cell
+- table-row
+- *other table elements...*
+
+
+- form
+- input
+- select
+- radio
+- checkbox
+- *other forms inputs...*
+
+### Tests
+
+#### Campaigns
+We currently have 2 campaigns implemented:
+
+- **Sanity**: its purpose is to validate a Pull Request. It is executed in a CI environment (on Travis CI) for every Pull Request and is mandatory, i.e. a failed test blocks the merge. It consists of a few tests of the core features of PrestaShop: orders in BO, Products in BO, catalog in FO, cart in FO, Checkout process in FO, and installation of the build.
+- **Functional**: it’s the biggest and most important campaign. Its purpose is to validate that every feature of PrestaShop works, by testing them one by one. It goes on every page and tests whatever it can: table (filtering, ordering, quick edits, etc), CRUD (Create Read Update Delete) of items (Orders, Customers, Products, and so on), settings changes…
+
+We plan on implementing 2 more campaigns:
+- **End to end**: its purpose is to check that popular user paths are working as intended. It will walk through the application and mimic a real user working on their daily routines as a merchant: checking products, generating invoices, creating a customer account and an order, adding a special voucher for a specific user, etc. The selected user paths will be chosen by the Product Team. Thanks to their merchants and agencies interviews, they have a pretty good idea of what a merchant does every day and how they use the software.
+- **Regression**: a test campaign tailored to only target majors and critical issues in the few last versions of PrestaShop, to make sure they don’t come back (hence the name).
+
+#### Mocha and Mochawesome
+Mocha (https://mochajs.org/) is our test runner (a framework that reads our test code and runs it). We use it in coordination with Mochawesome (https://www.npmjs.com/package/mochawesome), a plugin for Mocha. Mochawesome produces a full JSON report in addition to a beautiful HTML report (which we don’t really use). The JSON file is sent to our Nightly Board, it is then inserted into the database to let visitors browse reports and visualize statistics.
+
+#### Lambda function (Why ?)
+Using lambda functions in mocha is discouraged (https://mochajs.org/#arrow-functions). They bind `this` to the scope of the lambda function, making it impossible to use internal Mocha methods and objects. Since we use the Mocha context to store our Page Objects, we strongly advise you to use the normal `function()` syntax. 
```

Comment:
```suggestion
Using lambda functions in mocha is [discouraged](https://mochajs.org/#arrow-functions). They bind `this` to the scope of the lambda function, making it impossible to use internal Mocha methods and objects. Since we use the Mocha context to store our Page Objects, we strongly advise you to use the normal `function()` syntax. 
```